### PR TITLE
added else condition and changed "undefined" to undefined

### PR DESCRIPTION
**Pull request by hirentimbadiya**
added else condition and changed "undefined" to undefined for issue#4

issue : https://github.com/siddhigate/js-polyfills/issues/4

### CODE REVIEW

File: array-polyfills/forEach/forEach.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Array.prototype.myForEach = function (cb) {
   for (let i = 0; i < this.length; i++) {
-    if (typeof this[i] !== "undefined") {
+    if (this[i] === undefined) {
+      cb(undefined, i, this);
+    } else{
```
Comment (siddhigate) [inline array-polyfills/forEach/forEach.js:5]: ### For the following case:
```
const arr1 = [1, 2, 3, , , 6];
arr1.myForEach(el => console.log(el)); 
```
### Expected output is:
``` 
1 2 3 6
```
### But the output here is
```
1 2 3 undefined undefined 6
```

Comment (hirentimbadiya) [reply array-polyfills/forEach/forEach.js:5]: But in Javascript empty values have undefined data type

Comment (siddhigate) [reply array-polyfills/forEach/forEach.js:5]: Yes, that's the challenge. We have to differentiate between empty values and undefined values.

Comment (hirentimbadiya) [reply array-polyfills/forEach/forEach.js:5]: we cannot identify which are empty undefined or explicitly given undefined , we need to give explicitly null data type to empty values to identify it.

Comment (siddhigate) [reply array-polyfills/forEach/forEach.js:5]: `forEach` method does it. So we will need to find a solution to that.

**Pull request closed without merging**